src/components/tickets/useTableSort.ts

```diff
@@ -0,0 +1,44 @@
+import { fetchTicketAsync } from '../../store/tickets/thunk';
+import { useAppDispatch } from '../../store/hooks';
+import { useSearchParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+
+export const useTableSortAndFilter = () => {
+  const dispatch = useAppDispatch();
+
+  const [searchParams, setSearchParams] = useSearchParams();
+  const priority = searchParams.getAll('priority');
+  const _page = searchParams.get('_page') ?? '0';
+  const _limit = searchParams.get('_limit') ?? '8';
+
+  const [selectedPriorities, setSelectedPriorities] = useState<string[]>(priority);
+  const [paginationModel, setPaginationModel] = useState({
+    pageSize: +_limit,
+    page: _page === '0' ? +_page : +_page - 1,
```

Comment:
Below we have `params.set('_page', (paginationModel.page + 1).toString());` probably we don't need `+ 1` here in this case?

Reply:
No, we need '-1' in _page otherwise if we refresh the page - the page would gonbigger by 1 each time. And also we need +1 in params.set because the data from json will return the same for 0 and 1 page